src/main/java/org/jabref/logic/importer/fetcher/IacrEprintFetcher.java

```diff
@@ -0,0 +1,158 @@
+import java.nio.charset.StandardCharsets;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.jabref.logic.importer.FetcherException;
+import org.jabref.logic.importer.IdBasedFetcher;
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.fileformat.BibtexParser;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class IacrEprintFetcher implements IdBasedFetcher {
+
+    public static final String NAME = "IACR eprints";
+
+    private static final Log LOGGER = LogFactory.getLog(IacrEprintFetcher.class);
+    private static final Pattern DATE_FROM_WEBSITE_PATTERN = Pattern.compile("[a-z ]+(\\d{1,2} [A-Za-z][a-z]{2} \\d{4})");
+    private static final DateFormat DATE_FORMAT_WEBSITE = new SimpleDateFormat("dd MMM yyyy");
+    private static final DateFormat DATE_FORMAT_BIBTEX = new SimpleDateFormat("yyyy-MM-dd");
+    private static final Predicate<String> IDENTIFIER_PREDICATE = Pattern.compile("\\d{4}/\\d{3,5}").asPredicate();
+    private static final String CITATION_URL_PREFIX = "https://eprint.iacr.org/eprint-bin/cite.pl?entry=";
+    private static final String DESCRIPTION_URL_PREFIX = "https://eprint.iacr.org/";
+    private static final Charset WEBSITE_CHARSET = StandardCharsets.ISO_8859_1;
+
+    private final ImportFormatPreferences prefs;
+
+    public IacrEprintFetcher(ImportFormatPreferences prefs) {
+        this.prefs = prefs;
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        String identifierWithoutLettersAndSpaces = identifier.replaceAll("[^0-9/]", " ").trim();
+
+        if (!IDENTIFIER_PREDICATE.test(identifierWithoutLettersAndSpaces)) {
+            throw new FetcherException(Localization.lang("Invalid IACR identifier: '%0'.", identifier));
+        }
+
+        Optional<BibEntry> entry = createEntryFromIacrCitation(identifierWithoutLettersAndSpaces);
+
+        if (entry.isPresent()) {
+            setAdditionalFields(entry.get(), identifierWithoutLettersAndSpaces);
+        }
+
+        return entry;
+    }
+
+    private Optional<BibEntry> createEntryFromIacrCitation(String validIdentifier) throws FetcherException {
+        String bibtexCitationHtml = getHtml(CITATION_URL_PREFIX + validIdentifier);
+        String actualEntry = getValueBetween("<PRE>", "</PRE>", bibtexCitationHtml);
+
+        Optional<BibEntry> entry;
```

Comment:
I am not sure, but you probably need to initialize it with Optional.empty() or you could get still an NPE if no entry is found

Reply:
If the javadoc on BibtexParser.singleFromString is correct, it should always return an entry or an Optional.empty().
But I might as well initialize it...

Reply:
you can also just return the entry directly in the try construct. This is in my opinion the most readable solution.

Reply:
👍